src/http_server.c

```diff
@@ -0,0 +1,155 @@
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <string.h>
+#include <unistd.h>
+#include "config.h"
+#include "log.h"
+#include <pthread.h>
+#include "tunnel.h"
+#include <ctype.h>
+#include <signal.h>
+#include <netinet/tcp.h>
+
+
+
+int process_request(int new_socket, char *id) {
+    log_info("Processing request\n");
+    
+    int flag = 1;
+    setsockopt(new_socket, IPPROTO_TCP, TCP_NODELAY, &flag, sizeof(int));
+
+    const int buffer_size = 2048;
+    char *response[buffer_size];
+    memset(response,' ',buffer_size);
+    response[buffer_size - 1] = '\0';
+    char *reply = "HTTP/1.1 200 OK\r\n"
+            "Content-Type: text/plain\r\n"
+            "Connection: close\r\n\r\n"
+            "SAML Login...\r\n\0";
+    memcpy(response,reply,strlen(reply));
+
+    // Read the request
+    ssize_t write_result = write(new_socket, response, buffer_size);
+    (void)write_result;
+
+    // dup2(new_socket, STDOUT_FILENO);
+    // dup2(new_socket, STDERR_FILENO);
+
+
+
+    char request[1024];
+    ssize_t read_result = read(new_socket, request, sizeof(request));
+
+    // Check for '=id' in the response
+    if (read_result < 0 || strlen(request) < 10 || strncmp(request, "GET /?id=", 9) != 0) {
```

Comment:
If the received request from the server is larger than the buffer, the result will not be null-terminated, causing `strlen` to behave wrong.